matrix_sdk_base/src/client.rs

```diff
@@ -949,6 +970,64 @@ impl BaseClient {
         }
     }
 
+    /// Receive a successful /messages response.
+    ///
+    /// * `response` - The successful response from /messages.
+    pub async fn receive_messages(
+        &self,
+        room_id: &RoomId,
+        direction: &Direction,
+        response: &GetMessageEventsResponse,
+    ) -> Result<Vec<SyncRoomEvent>> {
+        let mut changes = StateChanges::default();
+
+        let mut events: Vec<SyncRoomEvent> = vec![];
+        for event in &response.chunk {
+            #[allow(unused_mut)]
+            let mut event: SyncRoomEvent = event.clone().into();
+
+            #[cfg(feature = "encryption")]
+            match hoist_room_event_prev_content(&event.event) {
+                Ok(e) => {
+                    if let AnySyncRoomEvent::Message(AnySyncMessageEvent::RoomEncrypted(
+                        encrypted,
+                    )) = e
```

Comment:
```suggestion
                Ok(AnySyncRoomEvent::Message(AnySyncMessageEvent::RoomEncrypted(
                    encrypted,
                )))
```

Maybe a bit easier to read :shrug: